infra/aws/terraform/prow-build-cluster/docs/NodeUpgradeRunbook.md

```diff
@@ -0,0 +1,31 @@
+# EKS Node Upgrades
+
+EKS provides [Managed Node Groups](https://docs.aws.amazon.com/eks/latest/userguide/update-managed-node-group.html) that are highly effective for rolling out upgrades. However, we encountered an issue where automatic updates fail due to a PodDisruptionBudget set on our Prow Jobs, resulting in a PodEvictionFailure error. To mitigate the risks associated with forcing the upgrade, which could potentially lead to orphaned test resources, we have devised a solution that involves the introduction of a secondary Node Group.
+
+**Problem:** The PodDisruptionBudget set on our Prow Jobs hinders the successful execution of automatic updates in EKS. As a consequence, we cannot perform updates seamlessly without encountering the PodEvictionFailure error.
```

Comment:
I'd clarify that the key issue is that EKS has a timeout of 15 minutes for eviction to be done, but jobs can take longer, up to 1+ hours.